### ISSUE TITLE
The training code realize

### ISSUE THREAD

**Issue by MinusOne2** [gangweix/pixel-perfect-depth#3]
@gangweix I implemented the training code myself, but unfortunately, the predicted depth maps are incorrect. Could you please help me identify the potential issues?

wo finetune

<img width="590" height="442" alt="Image" src="https://github.com/user-attachments/assets/4e1384ff-e602-4d54-9a15-5488850deba2" />

w finetune (few steps)

<img width="589" height="442" alt="Image" src="https://github.com/user-attachments/assets/2e32c339-20d8-4002-94f6-f2829507f903" />

Below are the key components of my implementation:

1. Depth Map Preprocessing (for Hypersim metric depth):

```
def preprocess_depth_map(self, 
                        depth_map: np.ndarray, 
                        epsilon: float = 1.0,
                        low_percentile: float = 2,
                        high_percentile: float = 98) -> np.ndarray:

    depth_log = np.log(depth_map + epsilon)
    d_min = np.percentile(depth_log, low_percentile)
    d_max = np.percentile(depth_log, high_percentile)
    depth_log = np.clip(depth_log, d_min, d_max)  # Optional: clamp values
    depth_normalized = (depth_log - d_min) / (d_max - d_min) - 0.5

    return depth_normalized
```

2. Forward Pass in PixelPerfectDepth:

```
def forward(self, image, depth):

    semantics = self.semantics_prompt(image).to(self.dtype)
    
    # Add noise to depth
    timestep_index = torch.randint(0, self.sampling_steps, [1])
    timestep = self.sampling_timesteps[timestep_index].to(self.device)
    noise = torch.randn_like(depth).to(self.dtype)
    latent = self.schedule.forward(depth, noise, timestep).to(self.dtype)

    with torch.autocast(device_type=self.device.type, dtype=self.dtype, enabled=True):
        cond = image - 0.5  # Normalize image: [0,1] -> [-0.5, 0.5]
        input = torch.cat([latent, cond], dim=1)
        pred = self.dit(x=input, semantics=semantics, timestep=timestep)  # Predicts velocity: (xT - x0)
        target = self.schedule.convert_to_pred(x_0=depth, x_T=noise, t=timestep, pred_type='velocity')

    return pred, target
```

Training Hyperparameters:
- input size [1024, 768]
- train_batch_size 4
- sampling_steps 4
